extension/WebJobs.Extensions.RabbitMQ/Trigger/RabbitMQListener.cs

```diff
@@ -20,126 +19,134 @@ namespace Microsoft.Azure.WebJobs.Extensions.RabbitMQ;
 
 internal sealed class RabbitMQListener : IListener, IScaleMonitor<RabbitMQTriggerMetrics>
 {
+    private const int ListenerNotStarted = 0;
+    private const int ListenerStarting = 1;
+    private const int ListenerStarted = 2;
+    private const int ListenerStopping = 3;
+    private const int ListenerStopped = 4;
+
     private const string RequeueCountHeaderName = "x-ms-rabbitmq-requeuecount";
 
+    private readonly IModel channel;
     private readonly ITriggeredFunctionExecutor executor;
+    private readonly ILogger logger;
     private readonly string queueName;
     private readonly ushort prefetchCount;
-    private readonly IRabbitMQService service;
-    private readonly ILogger logger;
-    private readonly string functionId;
-    private readonly IRabbitMQModel rabbitMQModel;
+    private readonly string logDetails;
 
-    private EventingBasicConsumer consumer;
+    private int listenerState = ListenerNotStarted;
     private string consumerTag;
-    private bool disposed;
```

Comment:
Replaced `disposed` and `started` with stricter thread-safe checks. This pattern can also be found in few other extensions.